www/class/centreonLDAP.class.php

```diff
@@ -819,6 +819,61 @@ private function getCnFromDn($dn)
         }
         return false;
     }
+
+    /**
+     * Set a relation between the LDAP's default contactgroup and the user
+     *
+     * @param int $arId
+     * @param int $contactId
+     *
+     * @return bool
+     */
+    public function setUserLdapDefautContactgroup(int $arId = null, int $contactId = null)
+    {
+        if (!$arId || !$contactId) {
+            return false;
+        }
+        $ldapCg = null;
+        try {
+            // searching the default contactgroup chosen in the ldap configuration
+            $resLdap = $this->db->prepare("SELECT ari_value FROM auth_ressource_info " .
+                "WHERE ari_name LIKE 'ldap_default_cg' AND ar_id = :arId");
+            $resLdap->bindValue(':arId', $arId, PDO::PARAM_INT);
+            $resLdap->execute();
+            while ($result = $resLdap->fetch()) {
+                $ldapCg = $result['ari_value'];
+            }
+            unset($resLdap);
+            if (null === $ldapCg) {
+                //no default contactgroup was set in the LDAP parameters
+                return true;
+            }
+
+            // checking if the user isn't already linked to this contactgroup
+            $resCgExist = $this->db->prepare("SELECT COUNT(*) AS `exist` FROM contactgroup_contact_relation " .
+                "WHERE contact_contact_id = :contactId AND contactgroup_cg_id = :ldapCg");
+            $resCgExist->bindValue(':contactId', $contactId, PDO::PARAM_INT);
+            $resCgExist->bindValue(':ldapCg', $ldapCg, PDO::PARAM_INT);
+            $resCgExist->execute();
+            $row = $resCgExist->fetch();
+            if ($row['exist'] != 0) {
+                //user already linked to this contactgroup
+                return true;
+            }
+            unset($resCgExist);
+
+            // inserting the user to the chosen default contactgroup
+            $resCg = $this->db->prepare("INSERT INTO contactgroup_contact_relation " .
+                "(contactgroup_cg_id, contact_contact_id) VALUES (:ldapCg, :contactId)");
+            $resCg->bindValue(':ldapCg', $ldapCg, PDO::PARAM_INT);
+            $resCg->bindValue(':contactId', $contactId, PDO::PARAM_INT);
+            $resCg->execute();
+            unset($resCg);
+        } catch (\PDOException $e) {
```

Comment:
In which case this exception happens ? 
It's bad practice to treat exceptionnal cases as alternative scenario for an algorithm.

Reply:
when you can't connet to the db